src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -87,14 +134,17 @@ def _restore_screenshot_directory(self):
         self.screenshot_root_directory = self._screenshot_path_stack.pop()
 
     def _get_screenshot_paths(self, filename):
-        if not filename:
-            self._screenshot_index += 1
-            filename = 'selenium-screenshot-%d.png' % self._screenshot_index
-        else:
-            filename = filename.replace('/', os.sep)
-
-        screenshotDir = self._get_screenshot_directory()
-        logDir = self._get_log_dir()
-        path = os.path.join(screenshotDir, filename)
-        link = robot.utils.get_link_path(path, logDir)
+        filename = filename.format(
+            index=self._get_screenshot_index(filename))
+        filename = filename.replace('/', os.sep)
+        screenshotdir = self._get_screenshot_directory()
+        logdir = self._get_log_dir()
+        path = os.path.join(screenshotdir, filename)
+        link = robot.utils.get_link_path(path, logdir)
```

Comment:
Oh, I seriously thought these two functions would be identical, but on a closer look there seem to be also other small differences. Please keep it like it is. Due to differences it is unlikely that `get_link_path` will be removed from Robot anyway.